### PR TITLE
Tooltips: Check if user is grounded for Terrain Pulse

### PR DESCRIPTION
**Pull request by Karthik99999**
https://www.smogon.com/forums/threads/bug-reports-v4-read-original-post-before-posting.3663703/post-8890803

### CODE REVIEW

File: src/battle-tooltips.ts [hunk 1]
```diff
@@ -1316,13 +1316,16 @@ class BattleTooltips {
 	 * Gets the proper current type for moves with a variable type.
 	 */
 	getMoveType(move: Move, value: ModifiableValue, forMaxMove?: boolean | Move): [TypeName, 'Physical' | 'Special' | 'Status'] {
+		const pokemon = value.pokemon!;
```
Comment (Zarel) [inline src/battle-tooltips.ts:1319]:
```suggestion
		const pokemon = value.pokemon;
```

Pretty sure this works without the `!`

**Review by Zarel — APPROVED** [review]
Everything else looks good.

File: src/battle-tooltips.ts (hunk 1 above)
Comment (Karthik99999) [reply src/battle-tooltips.ts:1319]: yeah you're right. actually there's a check to see if the variable is null later so it's better to not have it.

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**